### PR TITLE
Update samd21 usb

### PR DESCRIPTION
**Pull request by keithduncan**
I have a [usb serial echo example](https://gist.github.com/keithduncan/016c6c86b8397e0be63dcfc4468efe2c) working for my Arduino NB1500 using these changes.

I don’t have any of the Cortex-M0(+) boards in our example set to verify otherwise I’d contribute that gist as an example too.

### CODE REVIEW

File: hal/src/samd21/usb/bus.rs [hunk 1]
```diff
@@ -761,21 +711,24 @@ impl Inner {
             w.eorsm().set_bit();
             w.uprsm().set_bit()
         });
+
+        self.flush_eps();
```
Comment (keithduncan) [inline hal/src/samd21/usb/bus.rs]: This part is cargoculted from the samd51 usb module

File: hal/src/samd21/usb/bus.rs [hunk 2]
```diff
@@ -218,15 +168,15 @@ impl BufferAllocator {
...
         }
 
         let end_addr = unsafe { start_addr.offset(size as isize) };
         if end_addr > buf_end {
-            return Err(UsbError::SizeOverflow);
+            return Err(UsbError::EndpointMemoryOverflow);
         }
 
-        self.next_buf = unsafe { end_addr.offset_from(self.buffers.as_ptr()) as u16 };
+        self.next_buf = unsafe { end_addr.sub(self.buffers.as_ptr() as usize) as u16 };
```
Comment (keithduncan) [inline hal/src/samd21/usb/bus.rs]: This is also copied from the samd51 example and avoids using an unstable feature

File: hal/src/samd21/usb/bus.rs [hunk 3]
```diff
@@ -761,21 +711,24 @@ impl Inner {
             w.eorsm().set_bit();
             w.uprsm().set_bit()
         });
+
+        self.flush_eps();
+
+        self.usb().ctrlb.modify(|_, w| w.detach().clear_bit());
```
Comment (keithduncan) [inline hal/src/samd21/usb/bus.rs]: This part was preventing the device from showing up on my host's bus scan

File: hal/src/lib.rs [hunk 4]
```diff
@@ -64,8 +64,3 @@ pub use self::samd21::*;
 pub mod samd11;
 #[cfg(feature = "samd11c14a")]
 pub use self::samd11::*;
-
-#[cfg(all(feature = "usb", feature = "samd21"))]
```
Comment (keithduncan) [inline hal/src/lib.rs]: Looking for some :eyes: on this change, it appears to me that these are already re-exported by the family globs `samd21::*` etc

**Comment by sajattack** [comment]
LGTM

**Pull request merged**